educe/stac/graph.py

```diff
@@ -118,27 +115,21 @@ def recursive_cdu_heads(self, sloppy=False):
         A dictionary mapping each CDU to its recursive CDU
         head (see `cdu_head`)
         """
-        cache = {}
+        heads = {}
         def get_head(c):
-            if c in cache:
-                return cache[c]
-            else:
-                hd = self.cdu_head(c, sloppy)
-                if hd is None: return None
-                if self.is_cdu(hd):
-                    deep_hd = get_head(hd)
-                else:
-                    deep_hd = hd
-                if deep_hd is None:
-                    return None
-                else:
-                    cache[c] = deep_hd
-                    return deep_hd
+            if c in heads:
+                return heads[c]
+            hd = self.cdu_head(c, sloppy)
+            if (hd is not None) and self.is_cdu(hd):
+                hd = get_head(hd)
+            heads[c] = hd
+            return hd
+
         for c in self.cdus():
             get_head(c)
-        return cache
+        return heads
 
-    def without_cdus(self, sloppy=False):
+    def without_cdus(self, sloppy=False, mode='head'):
```

Comment:
Could I bother you into improving the existing docstring while you are at it, i.e. expand the docstring for this function to document both the existing `sloppy` and newly introduced `mode` keyword arguments?
e.g.
```
Parameters
---------------
sloppy: boolean, default=False
    blablabla

mode: string, default='head'
    Replacement strategy for edges related to CDUs. "head" blablabla, "broadcast" blablabla, "custom" blablabla.
"""